guides/rails_guides/markdown/renderer.rb

```diff
@@ -68,14 +68,14 @@ def convert_footnotes(text)
 
         def lexer_language(code_type)
           case code_type
-          when "css", "js", "html", "ruby", "sql", "yaml"
-            code_type
-          when "erb", "html+erb"
+          when "html+erb"
             "erb"
           when "bash"
             "console"
-          else
+          when nil
             "plaintext"
+          else
+            ::Rouge::Lexer.find(language) ? code_type : "plaintext"
```

Comment:
What if instead of checking for unknown lexers here, we handle them in `block_code`?

```suggestion
            code_type
```

```patch
diff --git a/guides/rails_guides/markdown/renderer.rb b/guides/rails_guides/markdown/renderer.rb
index 497e8076a6..cf644c6f23 100644
--- a/guides/rails_guides/markdown/renderer.rb
+++ b/guides/rails_guides/markdown/renderer.rb
@@ -13,7 +13,7 @@ class Renderer < Redcarpet::Render::HTML
       def block_code(code, language)
         formatter = Rouge::Formatters::HTML.new
         lexer = ::Rouge::Lexer.find_fancy(lexer_language(language))
-        code = formatter.format(lexer.lex(code))
+        code = formatter.format(lexer.lex(code)) if lexer
         <<-HTML
 <div class="code_container">
 <pre><code class="highlight #{lexer_language(language)}">#{code}</code></pre>
```

Reply:
@eugeneius That was the approach in 8f836c198e4bced587d921271ca87ba340e8a69e.  Personally, I feel like the current approach is cleaner.  To me, it makes sense for `lexer_language` to have the responsibility of returning a valid language.

Reply:
Making this change also means we are back to being able to pass through `console?lang=ruby`. I'm personally fine with either approach, I'd just like to be able to merge and move onto other changes. 😁

Reply:
@gerrywastaken  What would be the benefit of using a code fence like `console?lang=ruby` vs the approaches I mentioned in https://github.com/rails/rails/pull/39901#discussion_r460529136 ?

Reply:
@jonathanhefner Sorry I wasn't suggesting it would be a benefit over your suggestion. I should have worded that better. I meant we would need to take care to sanitise it before injecting it here: https://github.com/rails/rails/blob/c14da7be2216e59cca788f3c40843a91624b3762/guides/rails_guides/markdown/renderer.rb#L19

Personally I would prefer to leave the code as-is as it already takes care of this case.

Reply:
@gerrywastaken  Ah, I understand.  I believe we could use [`lexer.tag`](https://github.com/rouge-ruby/rouge/blob/v3.20.0/lib/rouge/lexer.rb#L225-L234) for that purpose.

Reply:
Okay, I withdraw my suggestion 😄 @jonathanhefner you did most of the review work here, so let's go with your preference.